processor/routingprocessor/internal/common/logger.go

```diff
@@ -0,0 +1,64 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/routingprocessor/internal/common"
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql"
+)
+
+type TQLLogger struct {
```

Comment:
I took it from the transform processor, probably worth extracting it to `pkg/telemetryquerylanguage` later?

Reply:
The reason this is in the transform processor instead of inside the TQL package is to keep the Zap dependency out of the TQL, which currently has a [fairly small number](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/90cc1c0b707c5d6c43e8bd6f9ea9208da17d332c/pkg/telemetryquerylanguage/go.mod#L5) of direct dependencies. If we want to share this definition between multiple packages, we could just put this in there and accept Zap as one additional dependency, or would need to create an additional package within the contrib repo to share this. @TylerHelmuth  thoughts on this?

Reply:
Based on the SIG meeting today we'll be switching the package to better natively support this kind of stuff, but that will take time.  I'd say for now, duplicating this is fine.  As an early adapter of the package you'll get to work with us as we adjust things haha in the future I think the logger will be provided for you.